Library/Homebrew/tab.rb

```diff
@@ -8,78 +8,41 @@
 require "extend/cachable"
 
 # Rather than calling `new` directly, use one of the class methods like {Tab.create}.
-class Tab
+class AbstractTab
   extend Cachable
 
   FILENAME = "INSTALL_RECEIPT.json"
 
-  # Check whether the formula was installed as a dependency.
+  # Check whether the formula or cask was installed as a dependency.
   #
   # @api internal
   attr_accessor :installed_as_dependency
 
-  # Check whether the formula was installed on request.
+  # Check whether the formula or cask was installed on request.
   #
   # @api internal
   attr_accessor :installed_on_request
 
-  # Check whether the formula was poured from a bottle.
-  #
-  # @api internal
-  attr_accessor :poured_from_bottle
-
-  attr_accessor :homebrew_version, :tabfile, :built_as_bottle,
-                :changed_files, :loaded_from_api, :time, :stdlib, :aliases, :arch, :source,
-                :built_on
-  attr_writer :used_options, :unused_options, :compiler, :source_modified_time
+  attr_accessor :homebrew_version, :tabfile, :loaded_from_api, :time, :arch, :source, :built_on
 
-  # Returns the formula's runtime dependencies.
+  # Returns the formula or cask runtime dependencies.
   #
   # @api internal
-  attr_writer :runtime_dependencies
+  attr_accessor :runtime_dependencies
 
-  # Instantiates a {Tab} for a new installation of a formula.
-  def self.create(formula, compiler, stdlib)
-    build = formula.build
-    runtime_deps = formula.runtime_dependencies(undeclared: false)
-    attributes = {
+  def self.generic_attributes(formula_or_cask)
+    {
       "homebrew_version"        => HOMEBREW_VERSION,
-      "used_options"            => build.used_options.as_flags,
-      "unused_options"          => build.unused_options.as_flags,
-      "tabfile"                 => formula.prefix/FILENAME,
```

Comment:
Might want to keep this attribute here? Ignore me if it's too painful to handle prefix/caskroom differences here.

Reply:
I left it out for now since the method being called is different for formulae casks (i.e. `formula.prefix` vs `cask.metadata_main_container_path`). If it's unclear, we could either set `tabfile` to `nil` here or maybe figure out a way to simplify this more